autopush/endpoint.py

```diff
@@ -254,6 +273,49 @@ def _db_error_handling(self, d):
         d.addErrback(self._response_err)
         return d
 
+    def _extract_jwt(self, token, crypto_key):
+        # first split and convert the jwt.
+        sig = None
+        payload = {}
+
+        def fix_padding(string):
+            return string + '===='[len(sig) % 4:]
+
+        try:
+            (ehead, epayload, sig) = token.split('.', 2)
+            sig = fix_padding(sig)
+        except ValueError:
+            raise VapidAuthException("Invalid bearer token")
+        head = json.loads(base64.urlsafe_b64decode(ehead))
+        payload = json.loads(base64.urlsafe_b64decode(epayload))
+        if sig and crypto_key:
+            key = base64.urlsafe_b64decode(fix_padding(crypto_key))
+            vk = ecdsa.VerifyingKey.from_string(key, curve=ecdsa.NIST256p)
+            try:
+                jws.verify(head, payload, sig, vk)
+            except Exception:
+                # failure. Probably should log this...
+                raise VapidAuthException("Invalid bearer token")
+        return payload
+
+    def _process_auth(self, result):
+        crypto_key = parse_header(
+            self.request.headers.get('crypto-key')).get('p256ecdsa')
+        authorization = self.request.headers.get('authorization')
+        if crypto_key and authorization:
+            (auth_type, token) = authorization.split(' ', 1)
+            # if it's a bearer token containing what may be a JWT
+            if auth_type.lower() == 'bearer' and '.' in token:
+                try:
+                    jwt = self._extract_jwt(token, crypto_key)
+                    self.aud = jwt.get('aud')
+                    self.exp = jwt.get('exp')
```

Comment:
Actually, I assume the verify function already verifies that. In which case, why do we need to log it?